src/serializer/serializer.js

```diff
@@ -66,18 +66,32 @@ export class Serializer {
   statistics: SerializerStatistics;
   react: ReactSerializerState;
 
+  _populateForbiddenNames(ast: BabelNode, realmPreludeGenerator: PreludeGenerator): void {
+    let forbiddenNames = realmPreludeGenerator.nameGenerator.forbiddenNames;
+
+    // by finding if an identifier is pure, we know if it's leaked to the global or not
+    // if it leaks, we make it fobidden to use
+    traverse(
```

Comment:
Ugh, this will make Prepack noticable slower... I'd like to avoid that for when niceness doesn't matter.

Reply:
I couldn’t think of another way of doing this properly.